controllers/postController.js

```diff
@@ -100,4 +102,159 @@ const list = async (req, res) => {
   }
 };
 
-module.exports = { upsert, list };
+const today = async (req, res) => {
+  if (!isValidString(req.params.keywordId) || isEmptyString(req.params.keywordId)) {
+    return res.status(400).send({ message: "[InvalidKeywordId] Error occured" });
+  }
+
+  const hasKeywordId = (await keywordModel.findOne({ _id: req.params.keywordId }).exec()) !== null;
+  if (!hasKeywordId) {
+    return res.status(400).send({ message: "[NotExistedKeywordId] Error occured" });
+  }
+
+  const keywordId = req.params.keywordId;
+
+  try {
+    const today = new Date();
+    const yesterday = new Date(today.getFullYear(), today.getMonth(), today.getDate() - 1);
+
+    const todayPostCount = await postModel
+      .find({ keywordId })
+      .find({
+        createdAt: {
+          $gte: today.setHours(0, 0, 0, 0),
+          $lt: today.setHours(23, 59, 59, 999),
+        },
+      })
+      .countDocuments()
+      .exec();
+    const yesterdayPostCount = await postModel
+      .find({ keywordId })
+      .find({
+        createdAt: {
+          $gte: yesterday.setHours(0, 0, 0, 0),
+          $lt: yesterday.setHours(23, 59, 59, 999),
+        },
+      })
+      .countDocuments()
+      .exec();
+
+    return res.status(200).json({
+      todayPostCount,
+      diffPostCount: todayPostCount - yesterdayPostCount,
+    });
+  } catch {
+    return res
+      .status(500)
+      .send({ message: "[ServerError] Error occured in 'postController.today'" });
+  }
+};
+
+const postCount = async (req, res) => {
+  if (!isValidString(req.params.keywordId) || isEmptyString(req.params.keywordId)) {
+    return res.status(400).send({ message: "[InvalidKeywordId] Error occured" });
+  }
+
+  const keywordInfo = await keywordModel.findOne({ _id: req.params.keywordId }).exec();
+  if (keywordInfo === null) {
+    return res.status(400).send({ message: "[NotExistedKeywordId] Error occured" });
+  }
+
+  const keywordId = req.params.keywordId;
+
+  let cursorId;
+  if (isValidString(req.query.cursorId)) {
+    cursorId = req.query.cursorId;
+  } else {
+    cursorId = new Date();
+    cursorId.setHours(0, 0, 0, 0);
+    cursorId.setDate(cursorId.getDate() - cursorId.getDay());
+  }
```

Comment:
문자열이 아닌 타입이 들어오는 경우, cursorId에 이번주 일요일에 대한 Date 객체를 넣는 것 같은데요, 지난 번에 내부 논의 후, 저희 명세서에서 `null`을 받는 경우를 다 `""(빈문자열)`을 받는 것으로 수정하였습니다.

```suggestion
  let cursorId;
  if (isValidString(req.query.cursorId)) {
    if (isEmptyString(req.query.cursorId)) {
      cursorId = new Date();
      cursorId.setHours(0, 0, 0, 0);
      cursorId.setDate(cursorId.getDate() - cursorId.getDay());
    } else {
      cursorId = req.query.cursorId;
    }
  } else {
    /* 에러 처리 */
  }
```

대략 위와 같은 형태가 될 수 있을 것 같은데, 확인 부탁드립니다. 아니면 이전과 같이 `!isValidString(req.query.cursorId)`에 대한 에러처리를 먼저 해도 좋을 것 같습니다.

cc @jin-ttao

Reply:
빈 문자열로 전달받았을 경우에 대해 잊고 있었네요..! 해당 로직 수정해서 반영했습니다!